### PR TITLE
Adding docs on adding new platforms and debugger types

### LINKED ISSUE #382: Add instructions on how to add new devices/targets/families to mbed-ls

### PR DESCRIPTION
**Pull request by bridadan**
Fixes #382 

Could you please review @cmonr and @theotherjimmy?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -366,3 +366,67 @@ $ mbedls
...
+- [DAPLink](https://github.com/ARMmbed/DAPLink)
+   - As well as the related but legacy [CMSIS-DAP](https://github.com/mbedmicro/CMSIS-DAP) firmware
+- ST-LINK
+- J-Link
+
+### Adding support for DAPLink-compatible platforms (DAPLink, ST-LINK, and CMSIS-DAP)
+
+Add an entry to the `daplink` section of the [`DEFAULT_PLATFORM_DB`](https://github.com/ARMmbed/mbed-ls/blob/master/mbed_lstools/platform_database.py#L45).
+
+If your platform's name is `NEW_PLATFORM` and it has platform ID of `9999`, the new entry should be:
```
Comment (cmonr) [inline README.md]: More for my info, what sets the platform ID?

File: README.md [hunk 2]
```diff
@@ -366,3 +366,67 @@ $ mbedls
...
+        ...
+    }
+}
+```
+
+Please order the entries by the platform ID when adding new platforms.
+
+### Adding support for J-Link platforms
+
+J-Link detection works a differently due to the information present on the platform's filesystem. All new entries should be added to the `jlink` section of the [`DEFAULT_PLATFORM_DB`](https://github.com/ARMmbed/mbed-ls/blob/master/mbed_lstools/platform_database.py#L45).
```
Comment (cmonr) [inline README.md]: "works a bit differently"

**Comment by cmonr** [comment]
@janjongboom Does this clarify things?

File: README.md (hunk 1 above)
Comment (bridadan) [reply README.md]: That stuff is defined the Mbed Enabled docs, but its the first 4 characters of the USB ID, which is set by the debug firmware (DAPLink, ST-Link, etc).

File: README.md (hunk 2 above)
Comment (bridadan) [reply README.md]: Oops, I accidentally a word 😄 I mean to say "J-Link detection work differently".

**Comment by bridadan** [comment]
@cmonr fixed!

**Pull request merged**